tooling/nargo_toml/src/lib.rs

```diff
@@ -391,10 +393,14 @@ pub enum PackageSelection {
 pub fn resolve_workspace_from_toml(
     toml_path: &Path,
     package_selection: PackageSelection,
+    current_compiler_version: Option<String>,
```

Comment:
Note: this also checks whether your dependencies are valid -- I have made this optional so as to allow a flag from nargo_cli to bypass this check and just compile dependencies incase a dependency accidentally specifies a specific version of the compiler instead of "any version above this version" and or if the user is sure that its safe to bypass this check